DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/src/CMakeLists.txt

```diff
@@ -0,0 +1,164 @@
+set(REGISTER_MAP_FUNCTOR_SOURCE_FILE register_map_functor_model.cpp)
+set(REGISTER_MAP_FUNCTOR_TARGET_NAME register_map_functor_model)
+set(REGISTER_MAP_FUNCTOR_EMULATOR_TARGET ${REGISTER_MAP_FUNCTOR_TARGET_NAME}.fpga_emu)
+set(REGISTER_MAP_FUNCTOR_SIMULATOR_TARGET ${REGISTER_MAP_FUNCTOR_TARGET_NAME}.fpga_sim)
+
+set(STREAMING_FUNCTOR_SOURCE_FILE streaming_functor_model.cpp)
+set(STREAMING_FUNCTOR_TARGET_NAME streaming_functor_model)
+set(STREAMING_FUNCTOR_EMULATOR_TARGET ${STREAMING_FUNCTOR_TARGET_NAME}.fpga_emu)
+set(STREAMING_FUNCTOR_SIMULATOR_TARGET ${STREAMING_FUNCTOR_TARGET_NAME}.fpga_sim)
+
+set(REGISTER_MAP_LAMBDA_SOURCE_FILE register_map_lambda_model.cpp)
+set(REGISTER_MAP_LAMBDA_TARGET_NAME register_map_lambda_model)
+set(REGISTER_MAP_LAMBDA_EMULATOR_TARGET ${REGISTER_MAP_LAMBDA_TARGET_NAME}.fpga_emu)
+set(REGISTER_MAP_LAMBDA_SIMULATOR_TARGET ${REGISTER_MAP_LAMBDA_TARGET_NAME}.fpga_sim)
+
+set(STREAMING_LAMBDA_SOURCE_FILE streaming_lambda_model.cpp)
+set(STREAMING_LAMBDA_TARGET_NAME streaming_lambda_model)
+set(STREAMING_LAMBDA_EMULATOR_TARGET ${STREAMING_LAMBDA_TARGET_NAME}.fpga_emu)
+set(STREAMING_LAMBDA_SIMULATOR_TARGET ${STREAMING_LAMBDA_TARGET_NAME}.fpga_sim)
+
+# FPGA device selection
+if(NOT DEFINED FPGA_DEVICE)
+    set(FPGA_DEVICE "Arria10")
+    message(STATUS "FPGA_DEVICE was not specified.\
+                    \nConfiguring the design to run on the default FPGA device ${FPGA_DEVICE} (Intel(R) Arria(R) 10 FPGAs). \
+                    \nPlease refer to the README for information on board selection.")
+else()
+    message(STATUS "Configuring the design to run on FPGA device ${FPGA_DEVICE}")
+endif()
+
+# This is a Windows-specific flag that enables exception handling in host code
+if(WIN32)
+    set(WIN_FLAG "/EHsc")
+endif()
+
+# This is for finding the install path to the necessary include files
+if(DEFINED ENV{INTELFPGAOCLSDKROOT})
+   set(SDK_ROOT_PATH $ENV{INTELFPGAOCLSDKROOT})
+else()
+   message(FATAL_ERROR "The INTELFPGAOCLSDKROOT environment variable must be defined for this tutorial to compile.")
+endif()
+
+# A DPC++ ahead-of-time (AoT) compile processes the device code in two stages.
+# 1. The "compile" stage compiles the device code to an intermediate representation (SPIR-V).
+# 2. The "link" stage invokes the compiler's FPGA backend before linking.
+#    For this reason, FPGA backend flags must be passed as link flags in CMake.
+set(EMULATOR_COMPILE_FLAGS "-fsycl -Wall -Wno-c++2b-extensions ${WIN_FLAG} -fintelfpga -DFPGA_EMULATOR -I${SDK_ROOT_PATH}/include/")
```

Comment:
Why are you using -Wno-c++2b-extensions?
I don't think any other sample uses it

Reply:
I think you can use `include_directories(${SDK_ROOT_PATH}/include/)` once and not have to add `-I${SDK_ROOT_PATH}/include/` to every target

Reply:
-Wno-c++2b-extensions is used because the interface annotations are using clang-annotate. This warning can be safely ignored. Also switched to use include_directories()